config/http_config.go

```diff
@@ -210,6 +210,12 @@ func (c *HTTPClientConfig) Validate() error {
 		if c.BasicAuth != nil {
 			return fmt.Errorf("at most one of basic_auth, oauth2 & authorization must be configured")
 		}
+		if len(c.OAuth2.ClientID) < 1 || (len(c.OAuth2.ClientSecret) < 1 && len(c.OAuth2.ClientSecretFile) < 1) {
+			return fmt.Errorf("the oauth2 client_id and either the client_secret or client_secret_file must be configured")
```

Comment:
I think that I would still prefer to use == 0 like elsewhere and have 2 different errors for better readability.